### ISSUE TITLE
Private key file throws error when trying load_pem_private_key

### ISSUE THREAD

**Issue by makiten** [box/box-python-sdk#136]
I'm running python-3.5.0 on OS X 10.11.5. When trying to use JWT Authorization, creating a JWTAuth object fails with:

```
Traceback (most recent call last):
  File "<stdin>", line 6, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/boxsdk/auth/jwt_auth.py", line 102, in __init__
    backend=default_backend(),
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/cryptography/hazmat/primitives/serialization.py", line 20, in load_pem_private_key
    return backend.load_pem_private_key(data, password)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/cryptography/hazmat/backends/multibackend.py", line 289, in load_pem_private_key
    return b.load_pem_private_key(data, password)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1069, in load_pem_private_key
    password,
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 1247, in _load_key
    mem_bio = self._bytes_to_bio(data)
  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/cryptography/hazmat/backends/openssl/backend.py", line 477, in _bytes_to_bio
    data_char_p = self._ffi.new("char[]", data)
TypeError: initializer for ctype 'char[]' must be a bytes or list or tuple, not str
```

The default `openssl` backend is what seems to be the problem, but I'm not sure if I'm missing a step. [This issue](https://github.com/pyca/pyopenssl/issues/15) suggests that this is an expected result.

I have tried a workaround that seems to work, but there are some issues there, as well.

**Comment by jmoldow** [comment]
What workaround did you try?

We could probably fix the problem by changing ``with open(rsa_private_key_file_sys_path) as key_file:`` to ``with open(rsa_private_key_file_sys_path, 'rb') as key_file:``, so that `.read()` returns `bytes` instead of `str`.

**Comment by makiten** [comment]
I encoded the data after opening the file. I noticed the `jwt_fixes` branch has `with open(rsa_private_key_file_sys_path, 'rb') as key_file:`, so I can try that and see if that passes the tests.

**Comment by makiten** [comment]
Okay, changing the code and the unit test to reflect the addition of `'rb'` got it to pass, so the `jwt_fixes` branch would work on my application.

However, my app is going to Bluemix, so I may have to manually install the SDK using `jwt_fixes`, since I just confirmed that Bluemix (or my runtime) will encounter the same error without using that branch.

**Comment by jmoldow** [comment]
I'll make a pull request with that fix, and do a release with it.

**Comment by jmoldow** [comment]
This should be fixed in v1.5.3: https://pypi.python.org/pypi/boxsdk/1.5.3
